parser-typechecker/src/Unison/Util/Text.hs

```diff
@@ -0,0 +1,133 @@
+{-# Language BangPatterns #-}
+{-# Language GeneralizedNewtypeDeriving #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Unison.Util.Text where
+
+import Data.String (IsString(..))
+import Data.Foldable (toList)
+import Data.List (unfoldr)
+import Prelude hiding (take,drop,replicate)
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+import qualified Unison.Util.Bytes as B
+import qualified Unison.Util.Rope as R
+
+-- Text type represented as a `Rope` of chunks
+newtype Text = Text (R.Rope Chunk) deriving (Eq,Ord,Semigroup,Monoid)
+
+data Chunk = Chunk {-# unpack #-} !Int {-# unpack #-} !T.Text
```

Comment:
Is this wrapper necessary? I'm curious if it would be feasible to add a size field to `One` on the Rope itself and just use that.
Are there other benefits to using Chunk that a `Rope Text` can't handle on its own?
Maybe `type Text = Rope Text` would suffice without the added indirection.

Reply:
You could add a size field to `One` (I had phrased it that way at one point) but then you have the size represented twice in the case where you have `Rope (Vector Word8)`.